doc/api.rst

```diff
@@ -0,0 +1,478 @@
+    # Arguments are labels for DPMF and MX objects.
+    input_master.mx["emission1"] = MX("biased", ["dist1", "dist2"])
+    input_master.mx["emission2"] = MX("biased", ["dist1", "dist2"])
+
+
+.. py:class:: DenseCPT
+
+    A Numpy ``ndarray`` representing a dense CPT with a specialized string method 
+    for writing to the parameter file. Supports up to 3 dimensional tables.
+    
+    .. py:method:: __init__(self, *args)
+
+        Create a :py:class:`DenseCPT` object storing the provided distribution.
+
+        :param args: The probability distribution as an array of probabilties which is interpreted by the Numpy ``array`` constructor. 
+        :type args: array_like
+
+    .. py:classmethod:: uniform_from_shape(*shape, self=0.0)
+
+        A class method for creating a :py:class:`DenseCPT` object with the provided 
+        shape.
+        If the table is 2 or 3 dimensional, the diagonal entries of the table 
+        are set to the ``self_transition`` parameter (default 0.0) and all other 
+        entries are set to be uniform. 
+
+        :param shape: Shape of Dense CPT table
+        :type shape: Array_like or multiple arguments
+        :param self_transition: Value for diagonal entries in the table. Defaults to 0.0
+        :type self: float
+
+Usage example:
+
+.. code-block:: python
+
+    # Create a custom DenseCPT in the InputMaster dense_cpt 
+    # InlineSection.
+    input_master.dense_cpt["start"] = \
+        DenseCPT([[0.7, 0.3], [0.8, 0.2]])
+    # Create a DenseCPT with specified diagonal value and 
+    # uniform other values
+    input_master.dense_cpt["transition"] = \
+        DenseCPT.uniform_from_shape(2, 2, self_transition = 0.6)
+
+
+.. py:class:: DeterministicCPT
+
+    A deterministic CPT described using an existing decision tree with a 
+    specialized string method for writing to the parameter file.
+
+    .. py:attribute:: cardinality_parents
+        :type: tuple[int]
+
+        A tuple of integers describing the cardinality (number of states) for
+        the parent variables. If it is empty, there are no parent variables.
+
+    .. py:attribute:: cardinality
+        :type: int
+
+        The cardinality of this variable.
+
+    .. py:attribute:: dt
+        :type: str
+
+        The name of the decision tree representing this deterministic CPT.
+
+    .. py:method:: __init__(self, cardinality_parents, cardinality, dt)
+        
+        Creates a :py:class:`DeterministicCPT` with the provided attributes.
+
+        :param cardinality_parents: The cardinality of parent variables
+        :type cardinality_parents: tuple[int] or tuple
+        :param cardinality: The cardinality of this variable
+        :type cardinality: int
+        :param dt: Name of an existing decision tree 
+        :type dt: str 
+
+
+Section Classes
+===============
+
+Classes to store multiple objects that form one section of the parameter file.
+These are used within the ``InputMaster`` object and need not be defined by 
```

Comment:
This is strange. Why are these documented then if no one who reads it is supposed to use it? I thought these would be useful in some weird unimplemented GMTK object corner cases?

Reply:
I'm unsure if these should be documented, as casual users will not need them but anyone extending the software will. I will discuss it at the meeting on Friday.